casepro/backend/junebug.py

```diff
@@ -463,3 +467,42 @@ def received_junebug_message(request):
         created_on=datetime.now(), has_labels=True)
 
     return JsonResponse(msg.as_json())
+
+
+@csrf_exempt
+def receive_identity_store_optout(request):
+    '''Handles optout notifications from the Identity Store.'''
+    if request.method != 'POST':
+        return JsonResponse({'reason': 'Method not allowed.'}, status=405)
+
+    try:
+        data = json.loads(request.body)
+    except ValueError as e:
+        return JsonResponse(
+            {'reason': 'JSON decode error', 'details': e.message}, status=400)
+
+    identity_id = data.get('identity_id')
+    optout_type = data.get('optout_type')
+
+    syncer = IdentityStoreContactSyncer()
+
+    # I don't know how to get the org
+    local_contact = syncer.fetch_local(org, identity_id)
+    if local_contact:
+        local_contact.lock(org, identity_id)
+        if optout_type == "forget":
+            local_contact.release()
+            # I only delete the local contact. Should I delete the remote one as
+            # well considering that IdentityStoreContact that isn't a model class?
+            return JsonResponse({"success": True})
+
+        elif optout_type == "stop" or optout_type == "stopall":
+            local_contact.is_blocked = True
+            return JsonResponse({"success": True})
+        elif optout_type == "unsubscribe":
+            # I don't know what to do here
```

Comment:
We also probably want a catch-all case at the end to handle cases we don't know about (perhaps  we should return an error response so that the identity store is aware that it sent it a kind of optout we don't know how to deal with yet).